extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/TokensHelper.java

```diff
@@ -77,7 +77,9 @@ public void accept(Tokens tokens, Throwable throwable) {
                     tokenRequestStateUpdater.set(TokensHelper.this, null);
                 }
             }
-        });
+        })
+                // prevent next subscriptions to trigger multiple times the HTTP request before the end of the first one
+                .memoize().indefinitely();
```

Comment:
Is this Uni reused or recreated every time? 

Without this information, this could have dramatic consequences.
Also, it may cause a duplicated context issue.

Reply:
Hi @cescoffier, in my understanding this Uni is created when a new request to get a new token is made and is replaced by the new token as soon as it's received. A new token is requested when the previous one expires every _N_ seconds.

Do you agree with that @sberyozkin ?

Reply:
@Dougniel Yes, I do, only a minor clarification is that a new token is requested if the current one expires not every N seconds, but only when it is found that the  current one has expired at  the moment it is required. But otherwise, indeed, Uni is only used to acquire new tokens, once the tokens are available, they are accessed directly, https://github.com/quarkusio/quarkus/blob/main/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/TokensHelper.java#L62

@cescoffier Does what @Dougniel explained and my comment above address your concern ?